torch/fx/passes/utils/matcher_utils.py

```diff
@@ -228,17 +228,43 @@ def _match_args(args1: Union[List, Tuple], args2: Union[List, Tuple]) -> bool:
 
             return True
 
-        match_found = match_found and _match_args(pn.args, gn.args)
-
-        pn_kwargs, gn_kwargs = [], []
-        if pn.kwargs.keys() == gn.kwargs.keys():
-            for key in pn.kwargs.keys():
-                pn_kwargs.append(pn.kwargs[key])
-                gn_kwargs.append(gn.kwargs[key])
+        # Flatten all args/kwargs into 1 list of args
+        pn_args, gn_args = None, None
+        if (
+            (len(pn.args) != len(gn.args) or len(pn.kwargs) != len(gn.kwargs)) and
+            pn.op == "call_function" and
+            isinstance(pn.target, torch._ops.OpOverload)
+        ):
+            args_schema = pn.target._schema.arguments
+
+            def get_all_arguments(orig_args, orig_kwargs):
+                all_args = []
+                for i, schema in enumerate(args_schema):
+                    if schema.name in orig_kwargs:
+                        all_args.append(orig_kwargs[schema.name])
+                    elif not schema.kwarg_only and i < len(orig_args):
+                        all_args.append(orig_args[i])
+                    else:
+                        all_args.append(schema.default_value)
+                return all_args
+
+            pn_args = get_all_arguments(pn.args, pn.kwargs)
+            gn_args = get_all_arguments(gn.args, gn.kwargs)
+
+        elif len(pn.args) == len(gn.args) and len(pn.kwargs) == len(gn.kwargs):
```

Comment:
nit: even if len(kwargs) are the same, people can specify kwarg for different things I think?
e.g. 
op1(x, b=1), op1(x, c=2)